### PR TITLE
fix(storage): avoid select unexpected sub level count

### PR DESCRIPTION
**Pull request by Li0k**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

https://github.com/risingwavelabs/risingwave/issues/11746

This pr minimizes the probability of exceptions by lowering the priority of unexpected programs.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by Little-Wallace — APPROVED** [review]
LGTM

File: src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs [hunk 1]
```diff
@@ -358,17 +358,38 @@ impl NonOverlapSubLevelPicker {
             return vec![];
         }
 
+        let mut expected = Vec::with_capacity(scores.len());
+        let mut unexpected = vec![];
+
+        for selected_task in scores {
+            if selected_task.sstable_infos.len() > MAX_COMPACT_LEVEL_COUNT {
```
Comment (Little-Wallace) [inline src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs:365]: Shall we do this optimization also for `max_file_number` ?

Comment (Li0k) [reply src/meta/src/hummock/compaction/picker/min_overlap_compaction_picker.rs:365]: I think file_count is easier to reach the limit than level_count, so we prioritize the level count, and then determine if we need to limit `max_file_number` based on subsequent performance.

**Review by zwang28 — APPROVED** [review]
LGTM

**Pull request merged**